src/modules/kvs/lookup.c

```diff
@@ -459,8 +478,12 @@ static lookup_process_t walk (lookup_t *lh)
                 continue;
             }
         }
-        else
+        else {
+            cache_entry_decref (wl->entry);
             wl->dirent = dirent_tmp;
+            wl->entry = entry;
+            cache_entry_incref (wl->entry);
```

Comment:
Really minor suggestion -- I'm just noticing that this code is repeated at least a few times:

```C
cache_entry_decref (wl->entry);
wl->dirent = dirent_tmp;
wl->entry = entry;
cache_entry_incref (wl->entry);
```

Maybe an inline convenience function, like `walk_level_update (walk_level_t *wl, json_t *dirent, struct cache_entry *entry)`?

Reply:
yeah, it was just at that threshold of "should have I have a convenience function"?  Since you noticed it, we'll say it crosses the threshold now.